### PR TITLE
add auto-import of gcsfs

### PR DESCRIPTION
**Pull request by mrocklin**
- [ ] Tests added / passed
- [x] Passes `flake8 dask`
- [x] Fully documented, including `docs/source/changelog.rst` for all changes
      and one of the `docs/source/*-api.rst` files for new API

### CODE REVIEW

**Comment by mrocklin** [comment]
cc @martindurant 

1.  Is there any way to test for this behavior?
2.  Is there a mature Azure solution that we can add here?  (no is fine, just curious)

**Comment by martindurant** [comment]
For Azure, I believe https://github.com/oxygenanywhere/dask/blob/master/dask/bytes/adl.py works, but it is not in a convenient location, and I don't know how we would test. Such a small piece of code could be in dask itself. Someone probably has Azure credentials, but there is no mocking library, only VCR, which is a nightmare.

**Comment by mrocklin** [comment]
@martindurant would you mind opening an issue for Azure support saying what would be necessary and pinging Travis to ping someone on the Microsoft side?  I suspect that he knows the right person to contact here.

File: dask/bytes/core.py [hunk 1]
```diff
@@ -483,6 +483,13 @@ def ensure_protocol(protocol):
                         "    or\n"
                         "    pip install s3fs")
 
+    elif protocol in ('gs', 'gcs'):
+        import_required('gcsfs',
```
Comment (martindurant) [inline dask/bytes/core.py]: The import will inject into the _filesystems dict - is that correct? The check is only really needed is gcs is not already in the dict, but I guess it has no overhead.

Comment (mrocklin) [reply dask/bytes/core.py]: I'm mostly just copying the behavior for `s3fs`.  This is the only place where I see that library listed explicitly.

Comment (martindurant) [reply dask/bytes/core.py]: OK, I'm happy with it.

**Pull request merged**